modules/codegen/src/smithy4s/codegen/transformers/AwsConstraintsRemover.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.codegen.transformers
+
+import software.amazon.smithy.build.ProjectionTransformer
+import software.amazon.smithy.build.TransformContext
+import software.amazon.smithy.model.Model
+
+/** A preprocessor that removes constraints from specifications of AWS. */
+class AwsConstraintsRemover extends ProjectionTransformer {
+
+  override def getName: String = AwsConstraintsRemover.name
+
+  private val constraintsToRemove = List("length", "range", "pattern")
+  private val awsNamespacePrefix = "com.amazonaws"
+
+  override def transform(context: TransformContext): Model = {
+    val transformer = context.getTransformer()
+    transformer.removeTraitsIf(
+      context.getModel(),
+      (shape, t) => {
+        val shapeName = t.toShapeId().getName()
+        shape.getId.getNamespace.startsWith(
+          awsNamespacePrefix
+        ) && constraintsToRemove.contains(shapeName)
+      }
+    )
```

Comment:
Is this how it should be done? The PR https://github.com/disneystreaming/smithy4s/pull/1110 was quite a lot more involved, so I'm not sure if I'm missing something crucial here.

Reply:
Well, in theory you should only remove them if they are used in output types, which does make it quite more involved. 

There's a construct provided by the "smithy-model" library called "NeighbourVisitor" or something, which allows to get a transitive closure from a shape in the model. 

So the algorithm that I'm looking for is : 

1. lookup for all operations in the aws namespace
2. map them to get their input shapes 
3. flatMap the input shapes with the neighbour visitor to transitively gather all shapes their are referencing 
4. remove the constraint traits on the set of shapes you get in 3

Reply:
I've gotten this:
```scala
        val shapeName = t.toShapeId().getName()
        val isOperation = shape.isOperationShape()
        val isInAwsNamespace =
          shape.getId.getNamespace.startsWith(awsNamespacePrefix)

        if (isOperation && isInAwsNamespace ) {
          val np = NeighborProvider.of(context.getModel())
          val neighbors = np.getNeighbors(shape).asScala.toList
        }
```

But I don't know Smithy so well that I can work the algorithm out... I've pushed the code I have so far, could use some more guidance on how to get it finished :P